addon/-private/system/relationships/has-many.js

```diff
@@ -105,6 +105,36 @@ import { DEBUG } from '@glimmer/env';
   You can also specify an inverse on a `belongsTo`, which works how
   you'd expect.
 
+  #### Sync relationships
+
+  Ember Data resolves sync relationships with the related resources
+  available in its local store, hence it is expected these resources
+  to be loaded before or along-side the primary resource.
+
+  ```app/models/post.js
+  import DS from 'ember-data';
+
+  export default DS.Model.extend({
+    comments: DS.hasMany('comment', {
+      async: false
+    })
+  });
+  ```
+
+  In contrast to async relationship, accessing a sync relationship
+  will never trigger a request to fetch the resources,
```

Comment:
Similar to the comment on `belongsTo`, this isn't strictly 100% true, but it is "true enough". There are situations involving reload in which we won't error and will fire a request.

Perhaps a better wording would be `will error on access when any of the known related resources have not been loaded.`